File: VB/PersonalOrganizer/Common/DataModel/IReadOnlyRepository.vb
Imports System
Imports System.Linq
Imports System.Collections.ObjectModel
Namespace Common.DataModel
    Public Interface IReadOnlyRepository(Of TEntity As Class)
        Function GetEntities() As IQueryable(Of TEntity)
        ReadOnly Property UnitOfWork As IUnitOfWork
        ReadOnly Property Local As ObservableCollection(Of TEntity)
    End Interface
End Namespace

File: CS/PersonalOrganizer/Common/DataModel/IReadOnlyRepository.cs
using System;
using System.Linq;
using System.Collections.ObjectModel;

namespace PersonalOrganizer.Common.DataModel {
    public interface IReadOnlyRepository<TEntity> where TEntity : class {
        IQueryable<TEntity> GetEntities();
        IUnitOfWork UnitOfWork { get; }
        ObservableCollection<TEntity> Local { get; }
    }
}
